File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        // Finding the tip point where the array is differentiated.
              int left = 0, right = nums.size() - 1;
        
        while (left < right) {
            int mid = left + (right - left) / 2;
            
            if (nums[mid] > nums[right]) {
                // Minimum is to the right of mid
                left = mid + 1;
            } else {
                // Minimum is at mid or to the left of mid
                right = mid;
            }
        }
        
        return nums[left];
    }
    
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        new_set=set()
        for i in nums:
            new_set.add(i)

        print(len(new_set))
        print(len(nums))
        if len(new_set)==len(nums):
            return False
        return True
        